component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java

```
@@ -2942,7 +2942,7 @@ private HibernateIdentityObject getHibernateIdentityObject(IdentityStoreInvocati
       {
 
          hibernateObject = (HibernateIdentityObject)hibernateSession.createCriteria(HibernateIdentityObject.class)
-            .add(Restrictions.eq("name", io.getName()))
+            .add(Restrictions.eq("name", io.getName()).ignoreCase())
```

Shouldn't it be an issue to use ignoreCase when exo is configured as case sensitive ? This could probably retrieve multiple users (same username but with different cases).

---

There is a test [here](https://github.com/exoplatform/gatein-portal/blob/5.1.0-RC02/component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java#L701) to ensure that the "case sensitive" option is considered, so even if the Hibernate query returns a result when case insensitive search is disabled, the option will be applied at this level.

---

My concern is about the uniqueResult clause in this query. If more than one result exists in the database (with different cases), it will throw an exception.

---

At first, I thought that it should throw an exception because it demonstrates that the IDM database isn't coherent and thus it must be fixed on database manually (no automatic fix could be done).
But, to ensure that sys admin could make a workaround by configuration, I made this search taking into account the configuration parameter to apply "ignoreCase" condition. Done.